cunumeric/module.py

```diff
@@ -5314,10 +5314,71 @@ def convolve(a, v, mode="full"):
     if mode != "same":
         raise NotImplementedError("Need to implement other convolution modes")
 
+    if a.ndim != v.ndim:
+        raise RuntimeError("Arrays should have the same dimensions")
+    elif a.ndim > 3:
+        raise NotImplementedError(f"{a.ndim}-D arrays are not yet supported")
+
     if a.ndim == 1 and a.size < v.size:
         v, a = a, v
 
-    return a.convolve(v, mode)
+    if a.dtype != v.dtype:
+        v = v.astype(a.dtype)
+    out = ndarray(
+        shape=a.shape,
+        dtype=a.dtype,
+        inputs=(a, v),
+    )
+    a._thunk.convolve(v._thunk, out._thunk, mode)
+    return out
+
+
+def cholesky(a, no_tril=False):
```

Comment:
I realized `cholesky` isn't one of the top-level functions, but is define in the `linalg` submodule. I guess we could move this function to there and rename it to `_cholesky`, as this one takes an extra keyword parameter (`no_tril`) that we don't want to expose.

Reply:
Note that `examples/cholesky` calls this version with `no_tril=True` (not the default), but the example seems to run if that is simply omitted

Reply:
https://github.com/nv-legate/cunumeric/pull/209/commits/7dce6ac8b53d4cde689e36b9d912e253410bd821